h2/src/test/org/h2/test/scripts/TestScript.java

```diff
@@ -87,7 +87,7 @@ public void test() throws Exception {
         if (config.networked && config.big) {
             return;
         }
-        reconnectOften = !config.memory && config.big;
+//        reconnectOften = !config.memory && config.big;
```

Comment:
I'd rather leave this in and split the script files whereever we change mode

Reply:
May be it's simpler to remember mode from the old connection and set it in a new connection during a reconnect or forbid reconnection if mode is not default like it is already forbidden if there are temporary tables?

Reply:
What about other non-persistent settings (there are quite a few, maybe not used in test yet), and session-level settings like "current schema"?